core/lib/spree/testing_support/factories/promotion_factory.rb

```diff
@@ -13,7 +13,17 @@
         Spree::Promotion::Actions::CreateItemAdjustments.create!(calculator: calculator, promotion: promotion)
       end
     end
+
+    trait :with_one_use_per_user_rule do
+      after(:create) do |promotion|
+        rule = Spree::Promotion::Rules::OneUsePerUser.create!
+        promotion.rules << rule
+        promotion.save!
```

Comment:
minor thing: this can be omitted as `<<` operator adds rule to the `promotion.rules` collection and persist them at the same time @KacperMekarski

Reply:
@nnande done.